zio-http/src/main/scala/zio/http/Http.scala

```diff
@@ -70,33 +96,46 @@ sealed trait Http[-R, +Err, -In, +Out] { self =>
   )(implicit trace: Trace): Http[R1, Err1, In1, Out1] =
     self.defaultWith(that)
 
+  final def catchAllDefect[R1 <: R, Out1 >: Out](f: Cause[Nothing] => ZIO[R1, Nothing, Out1])(implicit
```

Comment:
I think `catchAllCause` should let you get the defects. No need for a `catchAllDefect` method in particular.